src/chief-v2.sol

```diff
@@ -0,0 +1,100 @@
+pragma solidity >=0.5.0;
+
+import "ds-note/note.sol";
+
+contract TokenLike {
+    function transferFrom(address, address, uint) public returns (bool);
+}
+
+contract Chief is DSNote {
+    // --- Init ---
+    constructor(address gov_) public { 
+        gov = TokenLike(gov_); 
+        threshold = ONE / 2;
+    }
+
+    // --- Math ---
+    uint256 constant ONE = 10 ** 18;
+    function add(uint x, uint y) internal pure returns (uint z) {
```

Comment:
I liked the idea of defining the function that are used in the contract "above the fold" in some sense, but we can definitely switch to `ds-math` if people prefer.

Reply:
I see this pattern is used [elsewhere](https://github.com/makerdao/dss/blob/master/src/tune.sol#L71), but I'm not sure why. Gas-efficiency?

@rainbreak ?

Reply:
It's done this way in dss because 1) there is custom maths and 2) for absolute clarity about what is happening. In general I don't think it has to be done this way and inheriting from `DSMath` is ok here.